modules/rte_mis_mail/rte_mis_mail.module

```diff
@@ -0,0 +1,411 @@
+  if ($otp_in_database === $user_entered_otp) {
+    $otp_service->markEmailAsVerified($email);
+    // Enable form submission, mark account creation, and update form elements.
+    $form['actions']['submit']['#attributes']['disabled'] = FALSE;
+    $form_state->set('account_creation', TRUE);
+    $form['email_container']['account']['mail']['#attributes']['readonly'] = 'readonly';
+    unset($form['email_container']['verify_email']);
+    unset($form['email_container']['otp_container']);
+    $form_state->set('no_cache', TRUE);
+
+    // Get the renderer service.
+    $renderer = \Drupal::service('renderer');
+
+    // Command to update the submit button wrapper.
+    $response->addCommand(new ReplaceCommand(
+      '#account-creation-button',
+      $renderer->render($form['actions']['submit'])
+    ));
+
+    // Command to update the email container wrapper.
+    $response->addCommand(new ReplaceCommand(
+      '#email-container',
+      $renderer->render($form['email_container'])
+    ));
+    \Drupal::messenger()->deleteAll();
+    return $response;
+  }
+  // If OTP doesn't match,trigger an Ajax response and display an error message.
+  else {
+    $form_state->set('no_cache', TRUE);
+    $response->addCommand(new HtmlCommand('#email-validation-message', t('OTP did not match')));
+    \Drupal::messenger()->deleteAll();
+    return $response;
+  }
+
+}
+
+/**
+ * Implements hook_cron().
+ *
+ * For truncating all the rows in the database.
+ */
+function rte_mis_mail_cron() {
+  // Call the otp service.
+  $otp_service = \Drupal::service('rte_mis_mail.otp_service');
+  $fields = ['created', 'verified', 'email'];
+  // Storing the filtered emails.
+  $emails_to_preserve = $otp_service->filterRows($fields);
+  // Deleting all the mails excepts the stored one.
+  $otp_service->cleanUpOtps($emails_to_preserve);
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function rte_mis_mail_form_alter(&$form, FormStateInterface &$form_state, $form_id) {
+
+  if ($form_id == 'session_limit_form') {
+
+    if ($form_state->get('otp_generated')) {
+      $form['otp'] = [
+        '#type' => 'textfield',
+        '#title' => t('OTP'),
+      ];
+      $form['submit_otp'] = [
+        '#type' => 'submit',
+        '#value' => t('Submit OTP'),
+        '#submit' => ['rte_mis_mail_submit_otp'],
+        '#validate' => ['rte_mis_mail_validate_otp'],
+      ];
+      $form['resend_otp'] = [
+        '#type' => 'submit',
+        '#value' => t('Resend OTP'),
+        '#submit' => ['rte_mis_mail_resend_otp'],
+      ];
+      $form['submit']['#access'] = FALSE;
+    }
+    $form['#submit'][0] = 'rte_mis_mail_otp_session_limit_form_submit';
+  }
+}
+
+/**
+ * Custom submit handler for the session_limit_form form.
+ */
+function rte_mis_mail_otp_session_limit_form_submit(&$form, &$form_state) {
+
+  if (!$form_state->get('otp_generated')) {
+    rte_mis_mail_generate_otp($form, $form_state);
+  }
+}
```

Comment:
Why we need session limit form submit function?

Reply:
Thought of different section and measure, but I don't think we need it here.Hence, removed that.